File: TOC/mac_validator.py
# Write a program that validates the 48 bit MAC address


class Mac_Val:
    def __init__(self, mac_addr):
        self.mac_addr = mac_addr.lower()

    def is_mac_valid(self):
        if(len(self.mac_addr) > 12 or len(self.mac_addr) < 12):
            return False
        for x in range(len(self.mac_addr)):
            if not (self.mac_addr[x].isnumeric() or
                    (ord(self.mac_addr[x]) >= 97 and ord(self.mac_addr[x]) <= 102)):
                print(f'here {x}')
                return False
        return True


def main():
    mac = input("Enter the MAC Address: ")
    mac_check = Mac_Val(mac)
    if(mac_check.is_mac_valid()):
        print(f"'{mac}' is valid MAC address")
    else:
        print(f"'{mac}' is Invalid MAC address")


if __name__ == '__main__':
    main()

File: TOC/mac_validator.c
// Write a program that validates the 48 bit MAC address

#include <stdio.h>
#include <ctype.h>

int main()
{
    char mac_address[50];
    int j = 0, flag = 0;

    printf("\nEnter the mac_address: ");
    scanf("%[^\n]", mac_address);

    while (mac_address[j])
    {
        if (mac_address[j] >= 65 && mac_address[j] <= 90)
        {
            mac_address[j] = tolower(mac_address[j]);
        }

        if ((mac_address[j] >= 97 && mac_address[j] <= 102) ||
            (mac_address[j] >= '0' && mac_address[j] <= '9'))
        {
            flag = 1;
        }
        else
        {
            flag = 0;
            break;
        }
        j++;
        if (j > 12)
        {
            flag = 0;
            break;
        }
    }
    if (flag)
    {
        printf("\nValid MAC address\n");
    }
    else
        printf("\nInvalid MAC address\n");
}